### PR TITLE
Add docs in README and example/ for Fake.

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -278,6 +278,36 @@ cat.eatFood("Fish");
 await untilCalled(cat.eatFood(any)); // Completes immediately.
 ```
 
+## Writing a fake
+
+You can also write a simple fake class that implements a real class, by
+extending [Fake]. Fake allows your subclass to satisfy the implementation of
+your real class, without overriding each method with a fake; the Fake class
```
Comment (natebosch) [inline README.md]: how about "without overriding the methods that aren't used in your test"

Comment (srawlins) [reply README.md]: Done. Thanks.

**Comment by srawlins** [comment]
I think I like putting a Best Practices being in README. Its a great one-stop doc now, and I think that would be useful into, and would not bloat the doc.

**Comment by natebosch** [comment]
Sounds good. I'll take a first pass at writing this out and send something your way this afternoon.

**Pull request merged**